src/collection.ts

```diff
@@ -53,9 +59,358 @@ export abstract class ReadOnlyCollection<T extends TigrisCollectionType> impleme
 		this.config = config;
 	}
 
-	private isTxSession(txOrQuery: Session | unknown): txOrQuery is Session {
-		const mayBeTx = txOrQuery as Session;
-		return "id" in mayBeTx && mayBeTx instanceof Session;
+	/**
+	 * Inserts multiple documents in Tigris collection.
+	 *
+	 * @param docs - Array of documents to insert
+	 * @param tx - Session information for transaction context
+	 */
+	insertMany(docs: Array<T>, tx?: Session): Promise<Array<T>> {
+		return new Promise<Array<T>>((resolve, reject) => {
+			const docsArray = new Array<Uint8Array | string>();
+			for (const doc of docs) {
+				docsArray.push(new TextEncoder().encode(Utility.objToJsonString(doc)));
+			}
+
+			const protoRequest = new ProtoInsertRequest()
+				.setProject(this.db)
+				.setCollection(this.collectionName)
+				.setDocumentsList(docsArray);
+
+			this.grpcClient.insert(
+				protoRequest,
+				Utility.txToMetadata(tx),
+				(error: grpc.ServiceError, response: server_v1_api_pb.InsertResponse): void => {
+					if (error !== undefined && error !== null) {
+						reject(error);
+					} else {
+						const clonedDocs = this.setDocsMetadata(docs, response.getKeysList_asU8());
+						resolve(clonedDocs);
+					}
+				}
+			);
+		});
+	}
+
+	/**
+	 * Insert new or replace existing documents in collection.
+	 *
+	 * @param docs - Array of documents to insert or replace
+	 * @param tx - Session information for transaction context
+	 */
+	insertOrReplaceMany(docs: Array<T>, tx?: Session): Promise<Array<T>> {
+		return new Promise<Array<T>>((resolve, reject) => {
+			const docsArray = new Array<Uint8Array | string>();
+			for (const doc of docs) {
+				docsArray.push(new TextEncoder().encode(Utility.objToJsonString(doc)));
+			}
+			const protoRequest = new ProtoReplaceRequest()
+				.setProject(this.db)
+				.setCollection(this.collectionName)
+				.setDocumentsList(docsArray);
+
+			this.grpcClient.replace(
+				protoRequest,
+				Utility.txToMetadata(tx),
+				(error: grpc.ServiceError, response: server_v1_api_pb.ReplaceResponse): void => {
+					if (error !== undefined && error !== null) {
+						reject(error);
+					} else {
+						const clonedDocs = this.setDocsMetadata(docs, response.getKeysList_asU8());
+						resolve(clonedDocs);
+					}
+				}
+			);
+		});
+	}
+
+	/**
+	 * Inserts a single document in Tigris collection.
+	 *
+	 * @param doc - Document to insert
+	 * @param tx - Session information for transaction context
+	 */
+	insertOne(doc: T, tx?: Session): Promise<T> {
+		return new Promise<T>((resolve, reject) => {
+			const docArr: Array<T> = new Array<T>();
+			docArr.push(doc);
+			this.insertMany(docArr, tx)
```

Comment:
Given this is returning a promise, I think you could change this to:

```
async insertOne(doc: T, tx?: Session): Promise<T> {
		const docArr: Array<T> = [doc];			
                 await this.insertMany(docArr, tx)
                }
			

```

Reply:
insertMany returns `Promise<Array<T>>`, hence unwrapping the promise.